File: app-miyou/src/main/java/com/miyou/app/application/credit/usecase/CreditQueryUseCase.java
package com.miyou.app.application.credit.usecase;

import org.springframework.data.domain.Pageable;

import com.miyou.app.domain.credit.model.CreditTransaction;
import com.miyou.app.domain.credit.model.UserCredit;
import com.miyou.app.domain.dialogue.model.UserId;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

public interface CreditQueryUseCase {

	Mono<UserCredit> getBalance(UserId userId);

	Flux<CreditTransaction> getTransactions(UserId userId, Pageable pageable);
}

File: app-miyou/src/main/kotlin/com/miyou/app/application/credit/usecase/CreditQueryUseCase.kt
package com.miyou.app.application.credit.usecase

import com.miyou.app.domain.credit.model.CreditTransaction
import com.miyou.app.domain.credit.model.UserCredit
import com.miyou.app.domain.dialogue.model.UserId
import org.springframework.data.domain.Pageable
import reactor.core.publisher.Flux
import reactor.core.publisher.Mono

interface CreditQueryUseCase {
    fun getBalance(userId: UserId): Mono<UserCredit>

    fun getTransactions(
        userId: UserId,
        pageable: Pageable,
    ): Flux<CreditTransaction>
}
